src/library/export/libraryexporter.cpp

```diff
@@ -0,0 +1,60 @@
+#include "library/export/libraryexporter.h"
+
+#include <QProgressDialog>
+#include <QThreadPool>
+
+#include "library/export/engineprimeexportjob.h"
+#include "util/parented_ptr.h"
+
+namespace mixxx {
+
+LibraryExporter::LibraryExporter(QWidget* parent,
+        UserSettingsPointer pConfig,
+        TrackCollectionManager* pTrackCollectionManager)
+        : QWidget{parent},
+          m_pConfig{std::move(pConfig)},
+          m_pTrackCollectionManager{pTrackCollectionManager} {
+}
+
+void LibraryExporter::requestExportWithOptionalInitialCrate(
+        std::optional<CrateId> initialSelectedCrate) {
+    if (!m_pDialog) {
+        m_pDialog = make_parented<DlgLibraryExport>(
+                this, m_pConfig, m_pTrackCollectionManager);
+        connect(m_pDialog.get(),
+                SIGNAL(startEnginePrimeExport(EnginePrimeExportRequest)),
```

Comment:
Please use type-safe signal/slot connections of Qt 5

Reply:
Updated.